plugins/social_network/social_network.cpp

```diff
@@ -263,7 +259,26 @@ namespace golos { namespace plugins { namespace social_network {
 
     void social_network::impl::set_url(discussion& d) const {
         const comment_api_object root(database().get<comment_object, by_id>(d.root_comment));
-        d.url = std::string("/") + root.category + "/@" + root.author + "/" + root.permlink;
+
+        std::vector< std::string > tags;
+        if( root.json_metadata.size() ) {
+            try {
+                tags  = fc::json::from_string( root.json_metadata )["tags"].as< std::vector< std::string > >();
+                if(tags.at(0) != ""){
+                    d.url = "/" + tags.at(0) + "/@" + root.author + "/" + root.permlink;
+                }
+                else {
+                    d.url = "/" + tags.at(1) + "/@" + root.author + "/" + root.permlink;
+                }
```

Comment:
```
if(tags.at(0) != ""){
    d.url = "/" + tags.at(0) + "/@" + root.author + "/" + root.permlink;
}
else {
    d.url = "/" + tags.at(1) + "/@" + root.author + "/" + root.permlink;
}
```
It's better to rewrite this to `while` loop (in case of several `""` tags at beginning of `json_metadata.tags`).

Reply:
Agree. I recommend  to use the method `tags::get_metadata()`. It already filter empty tags, and it will not require `try-catch`